src/unit/test_sds.c

```diff
@@ -252,50 +252,55 @@ int test_sds(int argc, char **argv, int flags) {
     return 0;
 }
 
+size_t sdsRealAllocSize(sds s) {
+    return sdsAllocSize(s) + PREFIX_SIZE;
+}
+
 int test_typesAndAllocSize(int argc, char **argv, int flags) {
     UNUSED(argc);
     UNUSED(argv);
     UNUSED(flags);
 
     sds x = sdsnewlen(NULL, 31);
     TEST_ASSERT_MESSAGE("len 31 type", (x[-1] & SDS_TYPE_MASK) == SDS_TYPE_5);
+    TEST_ASSERT_MESSAGE("len 31 sdsAllocSize", sdsRealAllocSize(x) == s_malloc_size(sdsAllocPtr(x)));
     sdsfree(x);
 
     x = sdsnewlen(NULL, 32);
     TEST_ASSERT_MESSAGE("len 32 type", (x[-1] & SDS_TYPE_MASK) >= SDS_TYPE_8);
-    TEST_ASSERT_MESSAGE("len 32 sdsAllocSize", sdsAllocSize(x) == s_malloc_size(sdsAllocPtr(x)));
+    TEST_ASSERT_MESSAGE("len 32 sdsAllocSize", sdsRealAllocSize(x) == s_malloc_size(sdsAllocPtr(x)));
```

Comment:
s_malloc_size = zmalloc_size, but what we want to check is zmalloc_usable_size, right?

We can add a s_malloc_usable_size i sdsalloc.h?

```suggestion
    TEST_ASSERT_MESSAGE("len 32 sdsAllocSize", sdsAllocSize(x) == s_malloc_usable_size(sdsAllocPtr(x)));
```

Reply:
so damn right, what we want to check is actually zmalloc_usable_size.